src/components/policy/policy_regular/src/access_remote_impl.cc

```diff
@@ -0,0 +1,246 @@
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "policy/access_remote_impl.h"
+
+#include <algorithm>
+#include <iterator>
+#include "policy/cache_manager.h"
+#include "utils/logger.h"
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "PolicyManagerImpl")
+
+using policy_table::DeviceData;
+using policy_table::FunctionalGroupings;
+using rpc::policy_table_interface_base::EnumFromJsonString;
+
+namespace policy {
+
+struct ToHMIType {
+  policy_table::AppHMITypes::value_type operator()(int item) const {
+    policy_table::AppHMIType type = static_cast<policy_table::AppHMIType>(item);
+    if (!IsValidEnum(type)) {
+      LOG4CXX_WARN(logger_, "HMI type isn't known " << item);
+      type = policy_table::AHT_DEFAULT;
+    }
+    LOG4CXX_DEBUG(logger_,
+                  "HMI type: " << item << " - " << EnumToJsonString(type));
+    return policy_table::AppHMITypes::value_type(type);
+  }
+};
+
+struct Contained {
+ private:
+  const policy_table::Strings& params_;
+
+ public:
+  explicit Contained(const policy_table::Strings& params) : params_(params) {}
+  bool operator()(const RemoteControlParams::value_type& item) const {
+    return std::find_if(params_.begin(), params_.end(), CompareString(item)) !=
+           params_.end();
+  }
+  struct CompareString {
+   private:
+    const RemoteControlParams::value_type& value_;
+
+   public:
+    explicit CompareString(const RemoteControlParams::value_type& value)
+        : value_(value) {}
+    bool operator()(const policy_table::Strings::value_type& item) const {
+      return value_ == static_cast<std::string>(item);
+    }
+  };
+};
+
+struct ToModuleType {
+  std::string operator()(policy_table::ModuleTypes::value_type item) const {
+    policy_table::ModuleType type = static_cast<policy_table::ModuleType>(item);
+    return EnumToJsonString(type);
+  }
+};
+
+AccessRemoteImpl::AccessRemoteImpl() : cache_(new CacheManager()) {}
```

Comment:
@JackLivio RemoteControl requires module type managing by policies. This class implements this logic